ui/src/app/utils/access-utils.tsx

```diff
@@ -361,6 +361,52 @@ export const useIsUserDisabled = () => {
   return disabled;
 };
 
+export const useNeedsToAcceptTOS = () => {
+  const { authLoaded, isSignedIn } = useStore(authStore);
+  const [userRequiredToAcceptTOS, setUserRequiredToAcceptTOS] = useState<
+    boolean | undefined
+  >(false);
+  useEffect(() => {
+    if (!authLoaded) {
+      return;
+    }
+
+    let mounted = true;
+    if (!isSignedIn) {
+      setUserRequiredToAcceptTOS(false);
+    } else {
+      (async () => {
+        try {
+          await profileStore.get().load();
+          if (mounted) {
+            setUserRequiredToAcceptTOS(false);
+          }
+        } catch (e) {
+          const errorResponse = await convertAPIError(e);
+          if (errorResponse.statusCode === 401) {
+            setUserRequiredToAcceptTOS(true);
+          }
+        }
+      })();
+    }
+    return () => {
+      mounted = false;
+    };
+  }, [authLoaded, isSignedIn]);
+  return userRequiredToAcceptTOS;
+};
+
+export const acceptTermsOfService = (tosVersion) => {
+  (async () => {
+    try {
+      await profileApi().acceptTermsOfService(tosVersion);
+      window.location.reload();
```

Comment:
why do we reload here?

Reply:
we need to reload this because if the accept call is successful it just stays as is, it doesnt redirect to the original url it is supposed to go to